concepts/0420-rich-schemas-common/README.md

```diff
@@ -86,24 +103,45 @@ data registries (ledgers).
 
 ### Relationship
 - A credential definition refers to a single mapping object
-- A mapping object refers to 1 or more schema objects.
-Each attribute in a schema may be included in the mapping one or more times (it is possible to encode a single attribute 
+- A mapping object refers to a single schema object
+- If there is no existent single schema to be referenced by a mapping object,
+a new schema must be created potentially referencing or extending existing ones.
+- Each attribute in a schema may be included in the mapping one or more times (it is possible to encode a single attribute 
 in multiple ways). A mapping may map only a subset of the attributes of a schema.
-- A presentation definition refers to 1 or more schema and credential definition objects. A presentation definition may use only a
-subset of the attributes of a schema.  
-
+- A presentation definition refers to 1 or more schema, mapping or credential definition objects.
+ 
+A presentation definition may use only a subset of the attributes of a schema.  
+ 
 ![](relationship-diagram.png)
 
+### Usage of JSON-LD
+The following Rich Schema objects must be in JSON-LD format:
+- Schema
+- Mapping 
+- Presentation Definition
+
+Context object can also be in JSON-LD format.
+
+
+If a Rich Schema object is a JSON-LD object, the `content`'s `@id` field must be equal to the `id`.
+
+The only thing that we currently expect from json-ld processing is substitution of attributes by a fully-qualified ones.
+We may assume that contexts belonging to the current ledger only
```

Comment:
I'm curious to understand this a bit better. Does this mean that all contexts necessary to build and map a rich schema must be on the same ledger? Would this hold true for the use of schemas that are anchored in IPFS that sit alongside an IPFS implementation. If this is true, then does that mean that different implementations of sidetree could reuse contexts from "different registries"?

While I understand this question is a bit futuristic sounding, the main reason I ask is because it makes sense to design in a way that aligns with securekey's architecture which using sidetree with Fabric. For rich schemas, they wouldn't necessarily have to anchor them using the batch files in sidetree, and instead could use IPFS for this functionality.

Reply:
We have this assumption in Indy, where we assume all Schemas and corresponding Contexts should be on the same Ledger. BTW this assumption in Indy may be for the first MVP implementation of Rich Schema feature only.

As for the Aries side, I think I will remove this assumption.